src/sentry/api/endpoints/organization_metrics.py

```diff
@@ -95,6 +96,26 @@ def get_use_case_ids(request: Request) -> Sequence[UseCaseID]:
         )
 
 
+class OrganizationMetricsEnrollPermission(OrganizationPermission):
+    scope_map = {"PUT": ["org:read", "org:write", "org:admin"]}
+
+
+@region_silo_endpoint
+class OrganizationMetricsEnrollEndpoint(OrganizationEndpoint):
+    publish_status = {
+        "PUT": ApiPublishStatus.PRIVATE,
+    }
+    owner = ApiOwner.TELEMETRY_EXPERIENCE
+    permission_classes = (OrganizationMetricsEnrollPermission,)
+
+    """Enroll an organization in the custom metrics beta."""
+
+    def put(self, request: Request, organization) -> Response:
+        previous_state = organization.get_option("sentry:custom_metrics_access", False)
```

Comment:
Decided to give the option to invalidate the previous state in case subsequent calls are made.